synapseclient/models/table.py

```diff
@@ -353,8 +358,47 @@ class Table(TableSynchronousProtocol, AccessControllable):
             contain: letters, numbers, spaces, underscores, hyphens, periods, plus
             signs, apostrophes, and parentheses
         parent_id: The ID of the Entity that is the parent of this table.
-        columns: The columns of this table.
-        description: The description of this entity. Must be 1000 characters or less.
+        columns: The columns of this table. This is an ordered dictionary where the key is the
+            name of the column and the value is the Column object. When creating a new instance
+            of a Table object you may pass any of the following types as the `columns` argument:
+
+            - A list of Column objects
+            - A dictionary where the key is the name of the column and the value is the Column object
+            - An OrderedDict where the key is the name of the column and the value is the Column object
+
+            After the Table object is created the columns attribute will be an OrderedDict. If
+            you wish to replace the columns after the Table is constructed you may do so by
+            calling the `.set_columns()` method. For example:
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import Column, ColumnType, Table
+
+            syn = Synapse()
+            syn.login()
+
+            # Initialize the table with a list of columns
+            table = Table(name="my_table", columns=[Column(name="my_column", column_type=ColumnType.STRING)])
+
+            # Replace the columns with a different list of columns
+            table.set_columns(columns=[Column(name="my_new_column", column_type=ColumnType.STRING)])
+            table.store()
+            ```
+
+
+            The order of the columns will be the order they are stored in Synapse. If you need
+            to reorder the columns the recommended approach is to use the `.reorder_column()`
+            method. Additionally, you may add, and delete columns using the `.add_column()`,
+            and `.delete_column()` methods on your table class instance.
+
+
+            Note that the keys in this dictionary should match the column names as they are in
+            Synapse. However, know that the name attribute of the Column object is used for
+            all interactions with the Synapse API. The OrderedDict key is purely for the usage
+            of this interface. For example, if you wish to rename a column you may do so by
+            changing the name attribute of the Column object. The key in the OrderedDict does
+            not need to be changed. The next time you store the table the column will be updated
+            in Synapse with the new name and the key in the OrderedDict will be updated.
```

Comment:
Is there a reason we can't update the key when the name of the column is changed in the Column object it is mapped to? Or do you see it as advantageous to still be using the old key name?

It could be a little awkward if a user wants to perform multiple operations on the Table/Column before `store`-ing it again